init.lua

```diff
@@ -111,9 +111,67 @@ end
 
 
 --- Receiving ---
+-- IRCv3.2 tags
+local function parse_tags(tag_message)
+	local tags = {}
+	local is_on_name = true
+	local cur_name = ""
+	local escaping = false
+	local escapers = {
+		["s"] = " ",
+		["r"] = "\r",
+		["n"] = "\n",
+		[";"] = ";"
+	}
+	local charbuf = ""
```

Comment:
Never append to a string in a loop.
Build up a table and use table.concat at the end.